src/Microsoft.TestPlatform.CoreUtilities/Utilities/StringBuilderExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.CoreUtilities.Extensions
+{
+    using System;
+    using System.Text;
+
+    public static class StringBuilderExtensions
+    {
+        /// <summary>
+        /// Add double quote around string. Useful in case of path which has white space in between.
+        /// </summary>
+        /// <param name="result">string builder</param>
+        /// /// <param name="data">data to be appended.</param>
+        /// <returns></returns>
+
+        public static void AppendSafeWithNewLine(this StringBuilder result, string data)
+        {
+            if (!string.IsNullOrEmpty(data))
+            {
+                // Don't append more data if already reached max length.
+                if (result.Length >= result.MaxCapacity)
+                {
+                    return;
+                }
+
+                // Add newline for readbility.
+                data += Environment.NewLine;
+
+                // Append sub string of data if appending all the data exceeds max capacity.
+                if (result.Length + data.Length >= result.MaxCapacity)
+                {
+                    data = data.Substring(0, result.MaxCapacity - result.Length);
```

Comment:
What's the idea here, are we saying the original Sting content will remain, & only add small part of new data?

Reply:
Yes.